assets/js/modules/ads/components/PAXEmbeddedApp.js

```diff
@@ -0,0 +1,230 @@
+	);
+
+	const isAdBlockerActive = useSelect( ( select ) =>
+		select( CORE_USER ).isAdBlockerActive()
+	);
+
+	const instanceID = useInstanceId( PAXEmbeddedApp, 'PAXEmbeddedApp' );
+
+	const newPromise = async () => {};
+
+	const callbackWithArgs = useCallback( ( name, callback = newPromise() ) => {
+		return async ( ...args ) => {
+			await global.console.debug( `Callback for: ${ name }`, {
+				args,
+				return: callback,
+			} );
+
+			return callback;
+		};
+	}, [] );
+
+	const paxAppRef = useRef();
+
+	const elementID = useMemo( () => {
+		return `googlesitekit-pax-embedded-app-${ instanceID }`;
+	}, [ instanceID ] );
+
+	const paxConfig = useMemo( () => {
+		return {
+			...( global?._googlesitekitPAXConfig || {} ),
+			clientConfig: {
+				contentContainer: `#${ elementID }`,
+			},
+		};
+	}, [ elementID ] );
+
+	const paxServices = useMemo( () => {
+		return {
+			authenticationService: {
+				get: callbackWithArgs( 'authenticationService:get', {
+					accessToken:
+						global?._googlesitekitPAXConfig?.authAccess
+							?.oauthTokenAccess,
+					authuser: '0',
+				} ),
+				fix: callbackWithArgs( 'authenticationService:fix', {
+					retryReady: true,
+				} ),
+			},
+			businessService: {
+				getBusinessInfo: callbackWithArgs(
+					'businessService:getBusinessInfo',
+					{
+						businessName: siteName,
+						// Part of the API for PAX.
+						// eslint-disable-next-line sitekit/acronym-case
+						businessUrl: siteURL,
+					}
+				),
+				fixBusinessInfo: callbackWithArgs(
+					'businessService:fixBusinessInfo',
+					{
+						retryReady: true,
+					}
+				),
+			},
+			conversionTrackingService: {
+				getSupportedConversionLabels: callbackWithArgs(
+					'conversionTrackingService:getSupportedConversionLabels',
+					{
+						conversionLabels: [
+							'purchase',
+							'subscribe',
+							'add_to_cart',
+							'begin_checkout',
+							'book_appointment',
+							'contact',
+							'request_quote',
+							'phone_call_leads',
+							'get_directions',
+							'submit_lead_form',
+							'sign_up',
+						],
+					}
+				),
+			},
+			termsAndConditionsService: {
+				notify: callbackWithArgs(
+					'termsAndConditionsService:notify',
+					{}
+				),
+			},
+		};
+	}, [ callbackWithArgs, siteName, siteURL ] );
+
+	const launchPAXApp = useCallback( async () => {
+		try {
+			paxAppRef.current =
+				await global.google.ads.integration.integrator.launchGoogleAds(
+					paxConfig,
+					paxServices
+				);
+
+			onLaunch?.();
+		} catch ( error ) {
+			setLaunchError( error );
+			global.console.error(
+				'Google Ads Partner Experience Error:',
+				error
+			);
+		}
+
+		setIsLoading( false );
+	}, [ paxConfig, paxServices, onLaunch ] );
+
+	useInterval( () => {
+		if ( ! launchGoogleAdsAvailable || hasLaunchedPAXApp ) {
+			return;
+		}
+
+		if (
+			typeof global?.google?.ads?.integration?.integrator
+				?.launchGoogleAds === 'function'
+		) {
+			setLaunchGoogleAdsAvailable( true );
+		}
+	}, 50 );
```

Comment:
We should probably clear the interval once the `launchGoogleAdsAvailable` becomes available

```suggestion
	}, launchGoogleAdsAvailable ? null : 50 );
```

Reply:
Oops, yeah, I was testing some stuff and forgot to restore this 👍🏻